api/wasm/cpp/proxy_wasm_api.h

```diff
@@ -373,6 +373,12 @@ class Context : public ContextBase {
 
   // Called on individual requests/response streams.
   virtual void onCreate() {}
+
+  virtual FilterStatus onNewConnection() { return FilterStatus::Continue; }
+  virtual FilterStatus onDownstreamData(size_t, bool) { return FilterStatus::Continue; }
+  virtual FilterStatus onUpstreamData(size_t, bool) { return FilterStatus::Continue; }
+  virtual void onConnectionClosed() {}
```

Comment:
We need an indicator of the direction I think.  So either add some flag(s)/enum here or 2 calls.  Personally I would make it a flags argument with bit fields for local and remote in case some environment does event consolidation.

Reply:
Done. Added `on{Downstream,Upstream}ConnectionClose({Unknown,Local,Remote})`.